### DISCUSSION TITLE
Should we support `tsconfig.*.json` files as JSONC format

### DISCUSSION THREAD

**Discussion by iFwu** [biomejs/biome#4652] [Ideas]
Currently, only `tsconfig.json` is recognized as JSONC format in biome ([src](https://github.com/biomejs/biome/blob/516314c188f9a25f7ae4f1659446ceb3a604112c/crates/biome_json_syntax/src/file_source.rs#L134)). However, some projects use multiple tsconfig files, like:
```json
{
  "files": [],
  "references": [
    { "path": "./tsconfig.app.json" },
    { "path": "./tsconfig.node.json" }
  ],
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/*": ["src/*"]
    }
  }
}
```
In such cases, formatting other tsconfig files with comments (e.g., `tsconfig.app.json`, `tsconfig.node.json`) results in errors. It would be beneficial to support tsconfig.*.json files as JSONC format to accommodate these scenarios.

Would it be possible to add support for tsconfig.*.json files to be recognized as JSONC format? This enhancement would greatly help projects that use multiple tsconfig files.

**Comment by arendjr** [comment]
I think the request is reasonable.

It also makes me think we may want to introduce a config setting that can force certain files to be interpreted as a certain type 🤔

**Comment by Conaclos** [comment]
You could use `tsconfig.*.jsonc` instead of `tsconfig.*.json`?
I used this in some of my personal projects.

> It would be beneficial to support tsconfig.*.json files as JSONC format to accommodate these scenarios.

If we choose this route, I could include all files starting with `tsconfig` and ending with `.json`

**Reply by clearfram3** [reply]
Yea, I agree here: why act as if `.json` is `.jsonc` when it's so easy to change the file name?
